io_util.py

```diff
@@ -305,3 +305,56 @@ def untar_input_file(tar_file_path: str) -> str:
         tar.extractall(path=path, filter="data")  # type: ignore
         filename = tar.getmembers()[0].name
     return path + f"/{filename}"
+
+
+# makes sure the model is available locally, if not download it from S3, if that fails download from Huggingface
+def check_model_availability() -> bool:
+    logger = logging.getLogger(__name__)
+    if os.path.exists(cfg.FILE_SYSTEM.BASE_MOUNT_MODEL + "/model.bin"):
```

Comment:
My worry is still that a model.bin might be available in the model base mount, from an earlier run, which is not the same as the one specified in  cfg.WHISPER_ASR_SETTINGS.MODEL
I don't think your solution is solving that yet, right? Is there any way to determine what version the model.bin corresponds to?

Reply:
Yes, my solution is indeed not solving that, but I don't think there is a way for me to solve it. I cannot check to which version the model.bin corresponds to (as far as I know) and the other files needed (tokenizer, vocab, config) also do not contain any explicit info about the model used. I can add a warning in the README saying that the user should double check if they already have a model stored locally so that they don't end up using the wrong one, but other than that there isn't much I can do. And also, I give the user the freedom to choose a model version that doesn't need to match the version found locally (and they can also set the MODEL parameter to an empty string, in which case it will not try to download anything and will only look for the model locally)

Reply:
Maybe we should discuss this in a short meeting